openmls/src/group/mls_group/test_mls_group.rs

```diff
@@ -1168,20 +1215,104 @@ fn group_context_extensions_proposal() {
+        &mut commit.tls_serialize_detached().unwrap().as_slice(),
+    )
+    .unwrap();
+
+    // Craft a commit that has two GroupContextExtension proposals. This is forbidden by the RFC.
+    // Change the commit before alice commits, so alice's state is still in the old epoch and we can
+    // use her state to forge the macs and signatures
+    match &mut franken_commit.body {
+        frankenstein::FrankenMlsMessageBody::PublicMessage(msg) => {
+            match &mut msg.content.body {
+                frankenstein::FrankenFramedContentBody::Commit(commit) => {
+                    let second_gces = frankenstein::FrankenProposalOrRef::Proposal(
+                        frankenstein::FrankenProposal::GroupContextExtensions(vec![
+                            frankenstein::FrankenExtension::LastResort,
+                        ]),
+                    );
+
+                    commit.proposals.push(second_gces);
+                }
+                _ => unreachable!(),
+            }
+
+            let group_context = alice_group.export_group_context().clone();
+
+            let bob_group_context = bob_group.export_group_context();
+            assert_eq!(
+                bob_group_context.confirmed_transcript_hash(),
+                group_context.confirmed_transcript_hash()
+            );
+
+            let secrets = alice_group.group.message_secrets();
+            let membership_key = secrets.membership_key().as_slice();
+
+            *msg = frankenstein::FrankenPublicMessage::auth(
+                alice_provider,
+                group_context.ciphersuite(),
+                &alice_signer,
+                msg.content.clone(),
+                Some(&group_context.into()),
+                Some(membership_key),
+                // this is a dummy confirmation_tag:
+                Some(vec![0u8; 32].into()),
+            );
+        }
+        _ => unreachable!(),
+    }
+
+    // alice merges the unmodified commit
     alice_group
-        .merge_pending_commit(provider)
+        .merge_pending_commit(alice_provider)
         .expect("error merging pending commit");
 
+    let fake_commit = MlsMessageIn::tls_deserialize(
+        &mut franken_commit.tls_serialize_detached().unwrap().as_slice(),
+    )
+    .unwrap();
+
+    println!("setting disable verification flag...");
+    validation_skip_handle.disable_validation();
+
+    println!("reading flag right after setting...",);
+    crate::skip_validation::is_disabled::confirmation_tag();
+
+    let err = bob_group
+        .process_message(bob_provider, fake_commit.into_protocol_message().unwrap())
+        .expect_err("expected an error");
+    assert!(matches!(
+        err,
+        ProcessMessageError::InvalidCommit(
+            StageCommitError::GroupContextExtensionsProposalValidationError(
+                GroupContextExtensionsProposalValidationError::TooManyGCEProposals
+            )
+        )
+    ));
+    validation_skip_handle.enable_validation();
```

Comment:
Instead of this disable - run - enable flow, would it make sense for `validation_skip_handle` to expose a function that does that for you? I.e. pass in a function pointer that is run with the check disabled?

Reply:
I added that, but left the old API public in case it is needed in some edge cases. Now we need to clone the `fake_commit` because we can't move it into the `FnMut` closure, but I think that is fine. The alternative would be to make it `FnOnce`, but that seems too limited.